pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -346,44 +359,58 @@ def build_model(
                 name="intercept", **self.model_config["intercept"]["kwargs"]
             )
 
-            beta_channel = self.beta_channel_dist(
-                name="beta_channel",
-                **self.model_config["beta_channel"]["kwargs"],
-                dims=("channel",),
-            )
-            alpha = self.alpha_dist(
-                name="alpha",
-                dims="channel",
-                **self.model_config["alpha"]["kwargs"],
-            )
-            lam = self.lam_dist(
-                name="lam",
-                dims="channel",
-                **self.model_config["lam"]["kwargs"],
-            )
-
+            # Adstock
             channel_adstock = pm.Deterministic(
                 name="channel_adstock",
-                var=geometric_adstock(
-                    x=channel_data_,
-                    alpha=alpha,
-                    l_max=self.adstock_max_lag,
-                    normalize=True,
-                    axis=0,
-                ),
-                dims=("date", "channel"),
-            )
-            channel_adstock_saturated = pm.Deterministic(
-                name="channel_adstock_saturated",
-                var=logistic_saturation(x=channel_adstock, lam=lam),
+                var=self.lag_function.apply(data=channel_data_),
                 dims=("date", "channel"),
             )
+
+            # Saturation
             channel_contributions = pm.Deterministic(
                 name="channel_contributions",
-                var=channel_adstock_saturated * beta_channel,
+                var=self.sat_function.apply(data=channel_adstock),
                 dims=("date", "channel"),
             )
 
+            # beta_channel = self.beta_channel_dist(
+            #     name="beta_channel",
+            #     **self.model_config["beta_channel"]["kwargs"],
+            #     dims=("channel",),
+            # )
+            # alpha = self.alpha_dist(
+            #     name="alpha",
+            #     dims="channel",
```

Comment:
what up with this commented code XD?

Reply:
haha, I needed as reference, will be deleted for sure before merge!